python/tests/param/params_test.py

```diff
@@ -0,0 +1,73 @@
+# Copyright 2017 Databricks, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+from __future__ import absolute_import, division, print_function
+
+from collections import namedtuple
+# Use this to create parameterized test cases
+from parameterized import parameterized
+
+from sparkdl.param.converters import SparkDLTypeConverters
+
+from ..tests import PythonUnitTestCase
+
+TestCase = namedtuple('TestCase', ['data', 'description'])
+
+_shared_invalid_test_cases = [
+    TestCase(data=['a1', 'b2'], description='required pair but get single element'),
+    TestCase(data=('c3', 'd4'), description='required pair but get single element'),
+    TestCase(data=[('a', 1), ('b', 2)], description='only accept dict, but get list'),
+    TestCase(data={1: 'a', 2.0: 'b'}, description='wrong mapping type'),
+    TestCase(data={'a': 1.0, 'b': 2}, description='wrong mapping type'),
+]
+_col2tnsr_test_cases = _shared_invalid_test_cases + [
+    TestCase(data={'colA': 'tnsrOpA', 'colB': 'tnsrOpB'},
+             description='strict tensor name required'),
```

Comment:
'tensor name required' should be good enough, since it's not clear what "strict tensor name" is to everyone.